### PR TITLE
Add verify script for staging repo metadata files

### PR DESCRIPTION
**Pull request by liggitt**
helps ensure we don't miss required files as new staging repos are added

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
(the test run is intended to fail to make sure it is called by the verify job)

**Comment by liggitt** [comment]
correctly noticed a missing file:
```
I0824 03:27:10.749] +++ command: bash "hack/make-rules/../../hack/verify-staging-meta-files.sh"
I0824 03:27:10.759] staging/src/k8s.io/client-go/README.md does not exist.
I0824 03:27:10.767] +++ exit code: 1
I0824 03:27:10.771] +++ error: 1
I0824 03:27:10.954] FAILED   verify-staging-meta-files.sh	0s
```

File: hack/verify-staging-meta-files.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+staging_repos=($(ls "${KUBE_ROOT}/staging/src/k8s.io/"))
+
+expected_filenames=(
```
Comment (dixudx) [inline hack/verify-staging-meta-files.sh]: missing `.github/PULL_REQUEST_TEMPLATE.md`.

Comment (liggitt) [reply hack/verify-staging-meta-files.sh]: added

File: hack/verify-staging-meta-files.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+exceptions=(
+  client-go/README.md # client-go provides its own README
+)
+
+RESULT=0
+for repo in ${staging_repos[@]}; do
+  for filename in ${expected_filenames[@]}; do
+    if echo " ${exceptions[*]} " | grep " ${repo}/${filename} " >/dev/null; then
```
Comment (sttts) [inline hack/verify-staging-meta-files.sh]: nit: `grep -F`

File: hack/verify-staging-meta-files.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      continue
```
Comment (sttts) [inline hack/verify-staging-meta-files.sh]: continue or elif

**Comment by sttts** [comment]
a nit. Otherwise :+1:

File: hack/verify-staging-meta-files.sh (hunk 3 above)
Comment (liggitt) [reply hack/verify-staging-meta-files.sh]: not sure what this is asking for

File: hack/verify-staging-meta-files.sh (hunk 2 above)
Comment (liggitt) [reply hack/verify-staging-meta-files.sh]: fixed

**Comment by spiffxp** [comment]
Perfect is the enemy of good so I won't block.  Improvements would be: checking kubernetes-template-project instead of a hardcoded list, and checking the content (ie: code-of-conduct.md should be the same file everywhere)

/lgtm
/hold
As long as you feel comfortable saying this isn't going to result in `touch $all-those-files-it-said-i-needed`, feel free /hold cancel

**Comment by liggitt** [comment]
> As long as you feel comfortable saying this isn't going to result in touch $all-those-files-it-said-i-needed, feel free to /hold cancel

I think people will largely do the right thing and copy/pasta the files from another staging repo folder. Empty files would at least raise an eyebrow in review, rather than someone needing to notice the absence of files.

/hold cancel

**Pull request merged**